packages/@lwc/template-compiler/src/shared/ast.ts

```diff
@@ -504,11 +506,13 @@ export function isComment(node: BaseNode): node is Comment {
     return node.type === 'Comment';
 }
 
-export function isExpression(node: Expression | Literal): node is Expression {
-    return node.type === 'Identifier' || node.type === 'MemberExpression';
+export function isExpression(node: BaseNode | Literal): node is Expression {
+    return node.type !== 'Literal';
```

Comment:
In a previous iteration of this PR, `isExpression` had been changed to accept the template expressions flag as its second argument. This led to a deluge of one-line changes where the flag had to be passed in.

To avoid this, I've updated the implementation here. With the flag disabled, the node will have type `Identifer`, `MemberExpression`, or `Literal`. Thus this new condition should be equivalent. The condition also works for the flag-enabled scenario.